### DISCUSSION TITLE
How can I check the validity of my replacement strategies?

### DISCUSSION THREAD

**Discussion by saffyhala** [ChampSim/ChampSim#516] [Q&A] [answered]
Hi there,
I have implemented my own eviction strategies (i.e. LRU, FIFO) and want to check the validity of my implementations. Is there anyway I could possibly integrate unit tests? How does one usually check if their implementations are correct?

**Comment by ngober** [comment, marked as answer]
The best way is to write unit tests on your own. You could put your tests in the `/test/cpp/src/` directory and ChampSim's makefile should pick them up and run them alongside the other tests. But, the best way might be to develop and run tests locally. It would give you more control over how to run them.

**Reply by saffyhala** [reply]
Is there a certain strucutre/template that I can use for my unit tests on ChampSim? I'm fairly new to ChampSim so I'm a bit confused.

**Reply by ngober** [reply]
ChampSim uses Catch2 to run its tests. But, you should use any technique or framework that gives you confidence in your results.
